webrender/src/render_backend.rs

```diff
@@ -1042,6 +1078,12 @@ impl RenderBackend {
             return;
         }
 
+        if !transaction_msg.resource_updates.is_empty() {
+            // Set render_on_generate_frame for any existing documents
+            // if resource_updates is not empty, since the updates could affects to another documents.
```

Comment:
"could affects to another documents"

Reply:
Why do we draw the connection between resource updates and frame generation here?

Reply:
It's not changing whether we generate a frame, it's just changing if we call render() at that time.

Before this patch we unconditionally called render() before generating a frame, now we only do so if something relevant has changed that would a new render. I assume that a resource update is one of those things, but not 100% sure.

Reply:
do you mean "called render() *after* generating a frame"?

Reply:
On windows, we do not get render() for updating the video frame by [Bug 1477608](https://bugzilla.mozilla.org/show_bug.cgi?id=1477608) if video frame format is same. Then the content does not have apz change nor animation nor display list update, we do not need to call render() during generating frame.